test/libsolidity/SemanticTest.h

```diff
@@ -58,7 +60,13 @@ class SemanticTest: public SolidityExecutionFramework, public EVMVersionRestrict
 	/// Compiles and deploys currently held source.
 	/// Returns true if deployment was successful, false otherwise.
 	bool deploy(std::string const& _contractName, u256 const& _value, bytes const& _arguments, std::map<std::string, solidity::test::Address> const& _libraries = {});
+
+	void addBuiltin(std::string _module, std::string _function, std::shared_ptr<Builtin> _builtin);
+
 private:
+	// builtin functions
+	std::optional<bytes> builtinSmokeTest(FunctionCall const& call);
```

Comment:
Maybe it would be better that builtins can only be defined with the help of a behaviour.